datafusion/physical-expr/src/expressions/lead_lag.rs

```diff
@@ -151,11 +151,11 @@ fn shift_with_default_value(
         create_empty_array(value, array.data_type(), array.len())
     } else {
         let slice_offset = (-offset).clamp(0, value_len) as usize;
-        let length = array.len() - offset.abs() as usize;
+        let length = array.len() - offset.unsigned_abs() as usize;
```

Comment:
looking at this code I wonder if we need to check for `offset < 0` -- however, this PR seems like a definite improvement so 👍